src/Nevermind/Nevermind.JsonRpc/JsonRpcModelMapper.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Linq;
+using System.Numerics;
+using Nevermind.Core;
+using Nevermind.Core.Crypto;
+using Nevermind.JsonRpc.DataModel;
+using Block = Nevermind.JsonRpc.DataModel.Block;
+using Transaction = Nevermind.JsonRpc.DataModel.Transaction;
+using TransactionReceipt = Nevermind.JsonRpc.DataModel.TransactionReceipt;
+
+namespace Nevermind.JsonRpc
+{
+    public class JsonRpcModelMapper : IJsonRpcModelMapper
+    {
+        private readonly ISigner _signer;
+
+        public JsonRpcModelMapper(ISigner signer)
+        {
+            _signer = signer;
+        }
+
+        public Block MapBlock(Core.Block block, bool returnFullTransactionObjects)
+        {
+            return new Block
+            {
```

Comment:
let us discuss how to prevent NullReferenceExceptions, I would suggest that we review each other if we are using ?. and ?? consistently and do asertions and so on
maybe some Debug.Assert(is full info block)